src/main/java/org/sagebionetworks/bridge/validators/ConsentSignatureValidator.java

```diff
@@ -57,6 +60,15 @@ public void validate(Object target, Errors errors) {
                     String message = String.format(TOO_YOUNG, minAgeOfConsent);
                     errors.rejectValue("birthdate", message);
                 }
+
+                // Handle participants' birthdays on 2/29 of a leap year.
+                if (period.getYears() == minAgeOfConsent && now.getMonthOfYear() == LEAP_YEAR_MONTH &&
+                        birthdate.getMonthOfYear() == LEAP_YEAR_MONTH &&
+                        birthdate.getDayOfMonth() == LEAP_YEAR_DAY)
+                {
+                    String message = String.format(TOO_YOUNG, minAgeOfConsent);
+                    errors.rejectValue("birthdate", message);
+                }
```

Comment:
When we calculate dates like this it is easy to get it wrong, and the leaper issue might only be one special case where we're getting date calculations wrong. It is safer to use a date and time library (we use Joda). Something like this:

        int minAgeOfConsent = 18;
        LocalDate birthday = LocalDate.parse("2004-01-24"); // the value entered
        LocalDate today = LocalDate.now(); // not exact, it'll be today in UTC and the user probably isn't in UTC
        boolean oldEnough = birthday.plusYears(minAgeOfConsent).isBefore(today);
        assertTrue(oldEnough);

You could also test if they are equal (in addition to before) as when we say you are 18, we aren't specifying what time zone you're in. So it may be enough that it's your birthday today somewhere in the world.

Reply:
Hi Alex, this is to handle the bug listed here: https://sagebionetworks.jira.com/browse/BRIDGE-2136

```
int minAgeOfConsent = 18;
LocalDate birthday = LocalDate.parse("2004-02-29"); // the value entered
LocalDate now = LocalDate.parse("2022-02-28"); // not exact, it'll be today in UTC and the user probably isn't in UTC
boolean oldEnough = birthday.plusYears(minAgeOfConsent).isEqual(now) || birthday.plusYears(minAgeOfConsent).isBefore(now);
assertTrue(oldEnough);
```

The above test passes. However, in reality, participants with birthdate on 2/29/2004 are not turning 18 years old until 3/1/2022.

Reply:
Then let's note it in the code because this looks pretty hacky. I can see someone else trying to remove it. I see two issues filed with Joda about this calculation that are closed as "won't fix" for backwards compatibility reasons:

https://github.com/JodaOrg/joda-time/issues/295
https://github.com/JodaOrg/joda-time/issues/132